f5_heat/resources/f5_sys_iappcompositetemplate.py

```diff
@@ -125,13 +92,13 @@ def _add_optional_attr(self, iapp_dict):
         :returns: possibly modified dictionary
         '''
 
-        if self.properties[self.COMPOSITE_TEMPLATE][self.REQUIRES_MODULES]:
+        if self.properties[self.REQUIRES_MODULES]:
```

Comment:
I don't think this pattern will do what you intend.   I think you will get what you intend if you use the pattern:

`if self.REQUIRES_MODULES in self.properties:`   etc.